### DISCUSSION TITLE
I Have a Statistics CircularBuffer

### DISCUSSION THREAD

**Discussion by brianmichalk1** [rlogiacco/CircularBuffer#78] [Show and tell]
I have been using a modified version of this library I call CircularBufferStatistics.  The cost is order(1) for queue operations, with some of the reporting O(n), but most queries O(1).
Right now, I am merging your latest with my code.  Would you like a PR for this, or would you rather me keep a separate fork?

**Comment by rlogiacco** [comment]
It seems interesting! If it's a feature easily disabled by a switch or a parallel class almost completely swappable in and out I believe it can be a wonderful addition!
A PR is very welcome and, above all, thanks for sharing your code 🙏
